meetings/2024-12-11.md

```diff
@@ -0,0 +1,134 @@
+- MdJ: Maybe we should talk to Tim
+
+### Trusted applications
+* MdJ: The 'trusted applications' mechanism we implemented in [NSS v5](https://github.com/nodeSolidServer/node-solid-server/blob/v5.0.0/CHANGELOG.md#500) was always meant to be a stop-gap. As solidcommunity.net is switching from NSS to Pivot, we need something to replace it. We are not yet at the point where working implementations of SAI exist. Therefore, I'm thinking of adding an app launcher along the lines of https://solid.pondersource.com/, that [edits the WAC ACLs and type index files](https://github.com/inrupt/launcher-exploration/blob/master/src/services/preparePod/preparePodForApp.ts) to allow apps as you launch them.
+  - ... we don't have SAI working in CSS.
+* ep: How do you identify apps in WAC ACLs?
+* MdJ: Web apps use the WebID of the logged in user and the client id is the app URL.
+  - ... the WAC spec says that that should work
+  - ... I don't know how it's implemented in specific servers.
+  - ... The special thing use is that the client ID matches the domain of the redirect URL from OAuth.
+* eP: If that that uses ACL orgin but not as the orgin header, I think it's not currently defined that way in WAC.
+  - ... In Solid OIDC we have to 2 options currently. [...]
+  - ... There is no global ID identifier such as WebID. 
+  - ... We can pick it up at one of the next meetings.
+* MdJ: we should check the WAC spec to make sure it's clear. I don't think we should move away from dynamic registrations. That's because you can trust a random app, even if the server never saw that app before.
+* eP: if you use URLs as client IDs, you don't need to do dynamic registrations. https://datatracker.ietf.org/doc/draft-parecki-oauth-client-id-metadata-document/
+* MdJ: so it's dynamic, although it doesn't use the OIDC dynamic registration.
+  - ... I opened a PR on pivot to change a dialog, related to this conversation here.
+  - ... [from chat]: https://github.com/solid-contrib/pivot/pull/38
+* eP: I don't think r/w that data, but all the data you have access to. If you're in a social graph, interacting with other users, that's a problem.
+* Niko: yes, that's the main reason I came today, this subject is very interesting to me. I am local first oriented. URLs are tied to servers, but local first does not involve a server [...]
+  - ... I am very interested in identifying applications. I am thinking of taking a hash on the binary. 
+* MdJ: There is also the concept of an instance of an app running on a specific device.
+  - ... Because there is no point of entry, as there is in a normal web server, probably the permissions should travel with the data. I am giving access to this data, and here's how to request it in the future.
+* eP: E2EE comes into play too. Should we put this on next week's agenda? How do we follow up?
+* MdJ: I'm gonna do some implementing and some documentation.
+* ...: There's an ongoing topic, how do we trust apps in Solid.
```

Comment:
```suggestion
* Niko: Yes, that's the main reason I came today, this subject is very interesting to me. I am local-first oriented. URLs are tied to servers, but local-first does not involve a server [...]
  - ... I am very interested in identifying applications. I am thinking of taking a hash on the binary. 
* MdJ: There is also the concept of an instance of an app running on a specific device.
  - ... Because there is no point of entry, as there is in a normal web server, the permissions should probably travel with the data. I am giving access to this data, and here's how to request it in the future.
* eP: E2EE comes into play, too. Should we put this on next week's agenda? How do we follow up?
* MdJ: I'm gonna do some implementing and some documentation.
* ...: There's an ongoing topic, how do we trust apps in Solid.
```